smaug/operators/padding_op.h

```diff
@@ -0,0 +1,98 @@
+#ifndef _OPERATORS_PADDING_OP_H_
+#define _OPERATORS_PADDING_OP_H_
+
+#include "smaug/core/backend.h"
+#include "smaug/core/operator.h"
+#include "smaug/core/tensor.h"
+#include "smaug/core/workspace.h"
+#include <google/protobuf/repeated_field.h>
+using namespace google::protobuf;
+
+namespace smaug {
+
+/** \ingroup Operators
+ * \brief Pad a given tensor in  any number of dimensions with arbitrary size.
+ *
+ * This has a software-based implementation.
+ *
+ * @tparam Backend The Backend that sets Alignment.
+ */
+template <typename Backend>
+class PaddingOp : public Operator {
+   public:
+    PaddingOp(const std::string& name, Workspace* workspace)
+            : Operator(name, OpType::Padding, workspace) {
+        inputs.resize(kNumInputs, nullptr);
+        outputs.resize(kNumOutputs, nullptr);
+    }
+
+    /**
+     * Set the paddingSize of the Tensor along each dimension.
+     * The paddingSize is orgainized as <dim1_forward, dim1_backward, ...
+     * ,dimk_backward>
+     */
+    void setPaddingSize(RepeatedField<google::protobuf::int32> const& val) {
+        std::vector<int> paddingSize(val.begin(), val.end());
```

Comment:
Sorry I didn't make it clear in my comment. This defines a new variable. Can you copy from the repeated field like:
`paddingSize.assign(val.begin(), val.end());`